### PR TITLE
Add of processPipeline support + tests improvement

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: otredis.go [hunk 1]
```diff
@@ -2,34 +2,68 @@ package otredis
...
 
 	"github.com/go-redis/redis"
-	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
 )
 
-// WrapRedisClient adds opentracing measurements for commands and returns cloned client
-func WrapRedisClient(ctx context.Context, c *redis.Client) *redis.Client {
+func WrapWithOpenTracing(ctx context.Context, client *redis.Client) *redis.Client {
```
Comment (ledor473) [inline otredis.go]: We should probably keep the same function name (for better compatibility)

File: otredis.go [hunk 2]
```diff
@@ -2,34 +2,68 @@ package otredis
...
-	// clone using context
-	copy := c.WithContext(c.Context())
-	copy.WrapProcess(func(oldProcess func(cmd redis.Cmder) error) func(cmd redis.Cmder) error {
+func process(parentSpan opentracing.Span, opts *redis.Options) func(oldProcess func(cmd redis.Cmder) error) func(cmd redis.Cmder) error {
+	return func(oldProcess func(cmd redis.Cmder) error) func(cmd redis.Cmder) error {
 		return func(cmd redis.Cmder) error {
-			tr := parentSpan.Tracer()
-			sp := tr.StartSpan("redis", opentracing.ChildOf(parentSpan.Context()))
-			ext.DBType.Set(sp, "redis")
-			sp.SetTag("db.method", cmd.Name())
```
Comment (ledor473) [inline otredis.go:28]: Seems like that line does not exist anymore, we should keep it in `doSpan`

File: README.md [hunk 3]
```diff
@@ -1,16 +1,16 @@
 # opentracing go-redis
 
-[OpenTracing](http://opentracing.io/) instrumentation for [go-redis](https://github.com/go-redis/redis).
+[OpenTracing](http://opentracing.io/) instrumentation for [go-redis](https://github.com/go-redis/redis) with support of `process` and `processPipeline` commands.
 
 ## Install
 
 ```
-go get -u github.com/smacker/opentracing-go-redis
+go get -u github.com/ticketmaster/opentracing-go-redis
```
Comment (ledor473) [inline README.md]: Let's keep the original repo, that wont work well when the PR is accepted in that repo

File: README.md [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 ```
 
 ## Usage
 
-Clone redis client `c := otredis.WrapRedisClient(ctx, c)` with a span.
+Clone redis client `c := otredis.WrapWithOpenTracing(ctx, c)` with a span.
```
Comment (ledor473) [inline README.md]: Not needed anymore as the same name is used

File: otredis.go (hunk 1 above)
Comment (ghost) [reply otredis.go]: Yep, I see that after my 1st commit and already fixed it :smile:

File: README.md (hunk 3 above)
Comment (ghost) [reply README.md]: Allright

File: otredis.go (hunk 2 above)
Comment (ghost) [reply otredis.go:28]: You mean the one with db.method?

File: README.md (hunk 4 above)
Comment (ghost) [reply README.md]: :+1:

File: otredis.go (hunk 2 above)
Comment (ledor473) [reply otredis.go:28]: Yes

Comment (ghost) [reply otredis.go:28]: Ok, fixed

**Pull request merged**